### ISSUE TITLE
Code generator fails to parse duplicate unique constraint

### ISSUE THREAD

**Issue by codemonstur** [jOOQ/jOOQ#9210]
When I use the following SQL:
```
CREATE TABLE `table1` (
 `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
 `email` varchar(45) NOT NULL,
 PRIMARY KEY (`id`),
 UNIQUE KEY `email_UNIQUE` (`email`)
);

CREATE TABLE `table2` (
 `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
 `email` varchar(45) NOT NULL,
 PRIMARY KEY (`id`),
 UNIQUE KEY `email_UNIQUE` (`email`)
);
```
The code generator will report an error:
```
Caused by: org.jooq.exception.DataAccessException: SQL [create table "table2"("id" bigint not null auto_increment, "email" varchar(45) not null, primary key ("id"), constraint "email_UNIQUE" unique ("email"))]; Constraint "email_UNIQUE" already exists; SQL statement:
create table "table2"("id" bigint not null auto_increment, "email" varchar(45) not null, primary key ("id"), constraint "email_UNIQUE" unique ("email")) [90045-199]
	at org.jooq_3.12.1.H2.debug(Unknown Source)
<snip/>
	at org.jooq.meta.extensions.ddl.DDLDatabase.create0(DDLDatabase.java:166)
	... 8 more
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Constraint "email_UNIQUE" already exists; SQL statement:
create table "table2"("id" bigint not null auto_increment, "email" varchar(45) not null, primary key ("id"), constraint "email_UNIQUE" unique ("email")) [90045-199]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:573)
<snip/>
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:369)
	... 14 more
```
It looks like MySQL allows UNIQUE constraints to have the same name while in H2 they need a globally unique name.

**Comment by lukaseder** [comment]
Thank you very much for your report.

We'll try to fix this soon via a DDL simulation that does not depend on H2 behind the scenes: #7034

I'm closing this as a duplicate of #8824

**Comment by lukaseder** [comment]
A workaround is documented in #8824
